src/components/FileTreeNode/index.tsx

```diff
@@ -160,6 +158,26 @@ class FileTreeNodeBase<TreeNodeType> extends React.Component<PublicProps> {
           ))}
       </React.Fragment>
     );
+  };
+
+  render() {
+    const { version } = this.props;
+
+    return (
+      <LinterProvider version={version}>
+        {/*
+          props.children needs be an anonymous function so that the
+          shallow prop check in connect(LinterProvider) will always think
+          there is a new value for props.children.
+
+          Without a new value, LinterProvider will not re-render its
+          children often enough. For example, the uncontrolled ListGroup
+          components (which could be nested recursively) need to
+          re-render when their internal state changes.
+        */}
```

Comment:
hu oh, is that because of `Treefold`?

Reply:
Yes. We should be controlling it with Redux and that would probably solve it. I'm not totally sure, though. I tried to do `<LinterProvider key={isExpanded}` ...>` but that doesn't account for the expansion of nested nodes.

Reply:
I was wondering if we want to connect this tree in #185, maybe we should reopen the issue? Or maybe not if the problem only occurs in the test suite 🤷‍♂️

Reply:
This issue doesn't affect the test suite at all. Clicking on the menu had no affect before I switched to using an anonymous function. It appeared broken because `connect()` was preventing the component from re-rendering.

> I was wondering if we want to connect this tree in #185

Yes, we should. We moved quickly on getting the prototype put together (which is fine) but I feel strongly that all parts of the app should be controlled by Redux. If we don't control *all* of the app with Redux, we shouldn't even bother to use Redux.